pic-sure-resources/pic-sure-aggregate-data-sharing-resource/src/main/java/edu/harvard/hms/dbmi/avillach/AggregateDataSharingResourceRSV3.java

```diff
@@ -0,0 +1,836 @@
+        }
+    }
+
+    @POST
+    @Path("/query/sync")
+    @Override
+    public Response querySync(QueryRequest queryRequest) {
+        logger.debug("Calling Aggregate Data Sharing Resource querySync()");
+        checkQuery(queryRequest);
+
+        HttpResponse response = null;
+        try {
+            Object query = queryRequest.getQuery();
+            UUID resourceUUID = queryRequest.getResourceUUID();
+
+            JsonNode jsonNode = objectMapper.valueToTree(query);
+            if (!jsonNode.has("expectedResultType")) {
+                throw new ProtocolException(ProtocolException.MISSING_DATA);
+            }
+            String expectedResultType = jsonNode.get("expectedResultType").asText();
+
+            Set<String> allowedResultTypes = Set.of(
+                "COUNT", "CROSS_COUNT", "INFO_COLUMN_LISTING", "OBSERVATION_COUNT", "OBSERVATION_CROSS_COUNT", "CATEGORICAL_CROSS_COUNT",
+                "CONTINUOUS_CROSS_COUNT", "VARIANT_COUNT_FOR_QUERY", "AGGREGATE_VCF_EXCERPT", "VCF_EXCERPT"
+            );
+
+            if (!allowedResultTypes.contains(expectedResultType)) {
+                logger.warn("Incorrect Result Type: " + expectedResultType);
+                return Response.status(Response.Status.BAD_REQUEST).build();
+            }
+
+            response = getHttpResponse(queryRequest, resourceUUID, "/query/sync", properties.getTargetPicsureUrl());
+
+            HttpEntity entity = response.getEntity();
+            String entityString = EntityUtils.toString(entity, "UTF-8");
+            String responseString = entityString;
+
+            responseString = getExpectedResponse(expectedResultType, entityString, responseString, queryRequest);
+
+            // propagate any metadata from the back end (e.g., resultId)
+            if (response.containsHeader(QUERY_METADATA_FIELD)) {
+                Header metadataHeader = ((Header[]) response.getHeaders(QUERY_METADATA_FIELD))[0];
+                return Response.ok(responseString).header(QUERY_METADATA_FIELD, metadataHeader.getValue()).build();
+            }
+
+            return Response.ok(responseString).build();
+        } catch (IOException e) {
+            logger.error(e.getMessage(), e);
+            throw new ApplicationException("Error encoding query for resource with id " + queryRequest.getResourceUUID());
+        } catch (ClassCastException | IllegalArgumentException e) {
+            logger.error(e.getMessage());
+            throw new ProtocolException(ProtocolException.INCORRECTLY_FORMATTED_REQUEST);
+        } finally {
+            closeHttpResponse(response);
+        }
+    }
+
+    private HttpResponse getHttpResponse(QueryRequest queryRequest, UUID resourceUUID, String pathName, String targetPicsureUrl)
+        throws JsonProcessingException {
+        String queryString = objectMapper.writeValueAsString(queryRequest);
+        String composedURL = HttpClientUtil.composeURL(targetPicsureUrl, "/v3" + pathName);
```

Comment:
This is the only change from the original file other than the path of this resource